addons/website/models/website.py

```diff
@@ -1632,35 +1632,72 @@ def _trigram_enumerate_words(self, search_details, search, limit):
             fields = set(fields).intersection(model._fields)
 
             unaccent = get_unaccent_wrapper(self.env.cr)
-            similarities = [sql.SQL("word_similarity({search}, {field})").format(
-                search=unaccent(sql.Placeholder('search')),
-                # Specific handling for website.page that inherits its arch_db and name fields
-                # TODO make more generic
-                field=unaccent(sql.SQL("{table}.{field}").format(
-                    table=sql.Identifier((self.env['ir.ui.view'] if field == 'arch_db' or (field == 'name' and 'arch_db' in fields) else model)._table),
-                    field=sql.Identifier(field)
-                )) if not (self.env['ir.ui.view'] if field == 'arch_db' or (field == 'name' and 'arch_db' in fields) else model)._fields[field].translate else
```

Comment:
This was unnecessary complicated for no reason, as reading the `translate` property through the ORM `_fields` property, `inherits` fields can be accessed from the parent model directly :shrug: 
Just
```py
if not model._fields[field].translate
```